framework/src/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -435,22 +443,24 @@ private AnnotationMirror createBooleanAnnotationMirror(List<Boolean> values) {
 
     private List<Number> calculateNumericalUnaryOp(
             Node operand, NumericalUnaryOps op, TransferInput<CFValue, CFStore> p) {
-        List<? extends Number> lefts = getNumericalValues(operand, p);
         List<Number> resultValues = new ArrayList<>();
-        for (Number left : lefts) {
-            NumberMath<?> nmLeft = NumberMath.getNumberMath(left);
-            switch (op) {
-                case PLUS:
-                    resultValues.add(nmLeft.unaryPlus());
-                    break;
-                case MINUS:
-                    resultValues.add(nmLeft.unaryMinus());
-                    break;
-                case BITWISE_COMPLEMENT:
-                    resultValues.add(nmLeft.bitwiseComplement());
-                    break;
-                default:
-                    throw new UnsupportedOperationException();
+        if (!isIntRange(operand, p)) {
```

Comment:
I was surprised to see this, because you put a lot of effort into `Range`'s support for unary operators, but this code doesn't use that.  This means that all estimates are lost for expressions that are annotated as `@IntRange`.

This is related to the comment above, but there and here is issue is more serious than handling mixed-type expressions.  Even above, if two expressions both have `@IntRange` annotations, then there is no `@IntRange` estimate for their result.  (Or am I wrong about that?)